runners/direct-java/src/test/java/org/apache/beam/runners/direct/KeyedPValueTrackingVisitorTest.java

```diff
@@ -144,10 +154,17 @@ public void keyedInputWithKeyPreserving() {
                         WindowedValue.getValueOnlyCoder(
                             KvCoder.of(StringUtf8Coder.of(), VarIntCoder.of())))));
 
+    TupleTag<KeyedWorkItem<String, KV<String, Integer>>> keyedTag = new TupleTag<>();
     PCollection<KeyedWorkItem<String, KV<String, Integer>>> keyed =
         input
-            .apply(GroupByKey.<String, WindowedValue<KV<String, Integer>>>create())
-            .apply(ParDo.of(new ParDoMultiOverrideFactory.ToKeyedWorkItem<String, Integer>()))
+            .apply(new DirectGroupByKeyOnly<String, WindowedValue<KV<String, Integer>>>())
+            .apply(
+                new DirectGroupAlsoByWindow<String, WindowedValue<KV<String, Integer>>>(
+                    WindowingStrategy.globalDefault(), WindowingStrategy.globalDefault()))
+            .apply(
+                ParDo.of(new ParDoMultiOverrideFactory.ToKeyedWorkItem<String, Integer>())
```

Comment:
This is a good change. It tests only that the analysis works on the post-surgery graph containing real direct runner primitives.